ui/plugins/failure_summary_panel.jsx

```diff
@@ -0,0 +1,220 @@
+    </li>
+);
+
+
+const BugListItem = (props) => {
+    const pinboardServiceEvent = () => {
+        props.pinboardService.addBug(props.bug, props.selectedJob);
+    };
+
+    const getBugUrl = props.getBugUrl(props.bug.id);
+    const bugSummaryText = props.escapeHTMLFilter(props.bug.summary);
+    const bugSummaryHTML = { __html: props.highlightCommonTermsFilter(bugSummaryText, props.suggestion.search) };
+
+    return (
+        <li>
+            <button className="btn btn-xs btn-light-bordered"
+                    onClick={pinboardServiceEvent}
+                    title="add to list of bugs to associate with all pinned jobs">
+                <i className="fa fa-thumb-tack"></i>
+            </button>
+            <a className={props.bugClassName}
+               href={getBugUrl}
+               target="_blank">{props.bug.id}
+                <span className={props.bugClassName} dangerouslySetInnerHTML={bugSummaryHTML}></span>
+            </a>
+     </li>
+    );
+};
+
+
+const ErrorsList = (props) => {
+    const errorListItem = props.errors.map((error, index) =>
+        <li key={index}>{error.name} : {error.result}.
+            <a title="Open in Log Viewer"
+               target="_blank"
+               href={error.lvURL}><span className="ml-1">View log</span></a>
+        </li>);
+
+    return (
+        <li>
+            No Bug Suggestions Available.<br />
+            <span className="font-weight-bold">Unsuccessful Execution Steps</span>
+            <ul>{errorListItem}</ul>
+        </li>
+    );
+};
+
+
+class FailureSummaryPanel extends React.Component {
+    constructor(props) {
+        super(props);
+    }
+
+    render() {
+        const escapeHTMLFilter = this.props.$injector.get('$filter')('escapeHTML');
+        const highlightCommonTermsFilter = this.props.$injector.get('$filter')('highlightCommonTerms');
+
+        return (
+            <ul className="list-unstyled failure-summary-list">
+                {this.props.suggestions && this.props.suggestions.map((suggestion, index) =>
+                    <SuggestionsListItem
+                                    key={index} index={index} suggestion={suggestion} user={this.props.user}
+                                    filerInAddress={this.props.filerInAddress} fileBug={this.props.fileBug}
+                                    highlightCommonTermsFilter={highlightCommonTermsFilter}
+                                    escapeHTMLFilter={escapeHTMLFilter} getBugUrl={this.props.getBugUrl}
+                                    bugLimit={this.props.bugLimit} pinboardService={this.props.pinboardService}
+                                    selectedJob={this.props.selectedJob} />)}
+
+                {this.props.errors && this.props.errors.length > 0 &&
+                <ErrorsList errors={this.props.errors} />}
+
+                {(!this.props.tabs.failureSummary.is_loading && this.props.jobLogsAllParsed && this.props.bugSuggestionsLoaded &&
```

Comment:
Just to be clear: since these are all ``&&`` there is no need to isolate the logic with parens...